pkg/ring/kvstore.go

```diff
@@ -18,35 +19,56 @@ var inmemoryStore KVClient
 type KVClient interface {
 	CAS(ctx context.Context, key string, f CASCallback) error
 	WatchKey(ctx context.Context, key string, f func(interface{}) bool)
+	WatchPrefix(ctx context.Context, prefix string, f func(string, interface{}) bool)
 	Get(ctx context.Context, key string) (interface{}, error)
 	PutBytes(ctx context.Context, key string, buf []byte) error
 }
 
+// KVConfig is config for a KVStore currently used by ring and HA tracker,
+// where store can be consul or inmemory.
+type KVConfig struct {
+	Store  string       `yaml:"store,omitempty"`
+	Consul ConsulConfig `yaml:"consul,omitempty"`
+
+	Mock KVClient
+}
+
+// RegisterFlags adds the flags required to config this to the given FlagSet
+func (cfg *KVConfig) RegisterFlags(f *flag.FlagSet, prefix string) {
+	cfg.Consul.RegisterFlags(f, prefix)
+	if prefix == "" {
+		prefix = "ring"
+	}
+	prefix = prefix + "."
```

Comment:
This seems a it redundant - we should just expect any prefix to end with a `.`.  And can we not default to `ring`, but instead add that in the appropriate place?

Reply:
It looks like the `ring` prefix wasn't there before, and we don't want to rename flags (it'll break upgrades).

Reply:
Right, that's why I'm not defaulting prefix from `""` to `"ring"` until after we register the Consul flags.

```cfg.Consul.RegisterFlags(f, prefix)
	if prefix == "" {
		prefix = "ring"
	}```

Reply:
Right I get it now thanks.